script/worker

```diff
@@ -133,8 +132,8 @@ $worker->log_setup_info();
 sub catch_exit {
     my ($sig) = @_;
 
-    log_info("Quit due to signal $sig");
-    $worker->stop(WORKER_COMMAND_QUIT);
+    log_info("Received signal $sig");
+    $worker->handle_signal($sig);
 }
```

Comment:
As we do not know anymore what the signal will result in I suggest to delete the function completely and instead use the new method `$worker->handle_signal` as signal handler directly … or the other way around as `handle_signal` would only be called from here anyway. You might want to keep it in the worker though to simplify reaching it during test runs for higher coverage

Reply:
I want to keep it in the worker because then I can simply call `handle_signal` from within the tests. And no, `handle_signal` can not be assigned directly to the signal handler because it needs the worker as first argument.

Reply:
ok, then maybe `sub catch_exit { $worker->handle_signal }`? (or is it `sub catch_exit { $worker->handle_signal(@_) }`?)

Reply:
Ok, I make it an one-liner. (I actually already have it but I'm waiting to push because I want to check the new coverage report first.)